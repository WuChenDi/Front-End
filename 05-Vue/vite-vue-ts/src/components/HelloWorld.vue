<template>
  <h1>{{ msg }}</h1>
  <button @click="handleVuex">change vuex</button>
  <h2 v-if="state.user.loading">我是 vuex 控制值</h2>
  <ul>
    <li v-for="product in products" :key="product.id">{{ product.title }}</li>
  </ul>
</template>

<script setup lang="ts">
import { useStoreHooks } from "../hooks/useStore";
const { state, getters, commit, dispatch } = useStoreHooks();

defineProps({
  msg: String,
});

console.log("vuex: ", state.user);
// console.log("getters", getters.user);
// console.log("getters", getters.user?.isLogin);
console.log("getters", getters["user/isLogin"]);

const handleVuex = () => {
  console.log("handleVuex", state.user.loading);
  commit("user/GET_DATA", !state.user.loading);
  // dispatch("user/GET_DATA");
};

import useProducts from "../models/products";
const { products } = await useProducts();
console.log(products);
</script>

<style scoped>
a {
  color: #42b983;
}

label {
  margin: 0 0.5em;
  font-weight: bold;
}

code {
  background-color: #eee;
  padding: 2px 4px;
  border-radius: 4px;
  color: #304455;
}
</style>
