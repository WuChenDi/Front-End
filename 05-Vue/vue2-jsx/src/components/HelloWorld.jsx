import './index.module.css';

export default {
  name: 'HelloWorld',
  props: {
    msg: String
  },

  render() {
    const { msg } = this;

    return (
      <div class>
        <h1>{msg}</h1>
      </div>
    );
  }
};
