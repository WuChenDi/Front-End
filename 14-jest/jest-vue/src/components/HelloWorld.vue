<template>
  <div class="hello">
    <h1 class="test">{{ msg }}</h1>
  </div>
</template>

<script>
export default {
  name: "HelloWorld",
  props: {
    msg: String,
  },
};
</script>
