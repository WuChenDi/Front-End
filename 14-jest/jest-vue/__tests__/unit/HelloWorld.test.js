import Vue from "vue";
import { shallowMount } from "@vue/test-utils";
import HelloWorld from "@/components/HelloWorld.vue";

// describe("HelloWorld.vue", () => {
//   it("renders props.msg when passed", () => {
//     const root = document.createElement("div");
//     root.className = "root";
//     document.body.appendChild(root);
//     new Vue({
//       render: (h) =>
//         h(HelloWorld, {
//           props: {
//             msg: "new message",
//           },
//         }),
//     }).$mount(".root");
//     console.log(document.body.innerHTML);
//     expect(document.getElementsByClassName("hello").length).toBe(1);
//   });
// });

describe("HelloWorld.vue", () => {
  // it("renders props.msg when passed", async() => {
  //   const msg = "new message";
  //   const wrapper = shallowMount(HelloWorld, {
  //     propsData: { msg },
  //   });
  //   expect(wrapper.props("msg")).toEqual(msg);
  //   expect(wrapper.findAll(".test").length).toBe(1);
  //   // expect(wrapper.text()).toMatch(msg);
  //   await wrapper.setProps({ msg: "hello" });
  //   expect(wrapper.props("msg")).toEqual("hello");
  // });
  it("组件渲染正常", () => {
    const wrapper = shallowMount(HelloWorld, {
      propsData: { msg: "new message" },
    });
    expect(wrapper).toMatchSnapshot()
  });
});
