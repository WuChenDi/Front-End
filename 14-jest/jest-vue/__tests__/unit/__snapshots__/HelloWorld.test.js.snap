// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloWorld.vue 组件渲染正常 1`] = `
<div class="hello">
  <h1 class="test">new message</h1>
</div>
`;
